src/main/java/tictactoe/BoardConverter.java

```diff
@@ -11,38 +13,87 @@ public String toString(Marks[][] board) {
 
         for (Marks[] columns : board) {
             for (Marks mark : columns) {
-                str.append(createLine(idSpot, mark));
+                str.append(createSpot(idSpot, mark));
                 idSpot++;
             }
 
             if (++idLine != board.length) {
-                str.append("-----------\n");
+                str.append(repeatElement(totalCharBoardWidth(board), INTER_LINE));
+                str.append(NEW_LINE);
             }
         }
 
         return str.toString();
     }
 
-    private String createLine(int idSpot, Marks mark) {
-        return " " + markToDisplay(mark, idSpot) + rightEdgeSpot(idSpot);
+    private String createSpot(int idSpot, Marks mark) {
+        String centeredMark = center(correctMark(mark, idSpot), spotWidth(board));
+        return centeredMark + elementRightSpot(idSpot);
     }
 
-    private String rightEdgeSpot(int y) {
+    private final String CROSS = "X";
+    private final String ROUND = "O";
+    private String correctMark(Marks mark, int idSpot) {
+        if (mark == Marks.CROSS) {
+            return CROSS;
+        } else if (mark == Marks.ROUND) {
+            return ROUND;
+        } else {
+            return String.valueOf(idSpot);
+        }
+    }
+
+    private final String SPOT_BORDER = "|";
+    private String elementRightSpot(int y) {
         if (isBoardEdge(y)) {
-            return " \n";
+            return NEW_LINE;
         } else {
-            return " |";
+            return SPOT_BORDER;
         }
     }
 
-    private char markToDisplay(Marks mark, int idSpot) {
-        if (mark == Marks.CROSS) {
-            return 'X';
-        } else if (mark == Marks.ROUND) {
-            return 'O';
+    private int spotWidth(Marks[][] board) {
+        int idMaxWidth = String.valueOf(board.length * board.length - 1).length();
+        idMaxWidth += 2;
+        return idMaxWidth;
+    }
+
+    private int totalCharBoardWidth(Marks[][] board) {
```

Comment:
I always try to avoid abbreviations in variable, class or method names.
It usually makes it easier for the readier. So `Character` instead of just `Char`, etc.

Thing is `totalCharBoardWidth` doesn't have a nice "readability factor" I think.
Try thinking in the way you use (as in _call_) the method and how it reads there.
A little further up that method is used in 
```
repeatElement(totalCharBoardWidth(board), INTER_LINE)
```

with some renaming it could read like

```
repeatElement(widthOf(board), LINE_SEPARATOR)
```